### PR TITLE
[Evals] CLI for scoring custom datasets

### PR DESCRIPTION
**Pull request by yanxi0830**
# TL;DR
- Update dataset register CLI to have flexibility to take input files
- Run app eval on custom dataset
  -  (1) via `--dataset-path` to read rows locally
  -  (2) via `--dataset-id` to read

image

# Test
#### Run on custom dataset
```
llama-stack-client eval run_scoring braintrust::answer-correctness braintrust::factuality \
--dataset-path <path-to-local-dataset> \
--output-dir ./ \
--num-examples 5
```

#### 1. Register a custom dataset
```
llama-stack-client datasets register --dataset-id <custom-dataset-id> \
--schema '{"generated_answer": {"type": "string"}, "expected_answer": {"type": "string"}, "input_query": {"type": "string"}}' \
--dataset-path <local-file-path> \
--provider-id huggingface-0
```
#### 2. Run scoring on custom dataset
```
llama-stack-client eval run_scoring \
braintrust::answer-correctness braintrust::factuality \
--dataset-id <custom-dataset-id> --output-dir ./ 
```

### CODE REVIEW

File: src/llama_stack_client/lib/cli/datasets/register.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    ctx,
+    dataset_id: str,
+    provider_id: Optional[str],
+    provider_dataset_id: Optional[str],
+    metadata: Optional[str],
+    url: str,
+    dataset_path: Optional[str],
+    schema: str,
+):
+    """Create a new dataset"""
```
Comment (ashwinb) [inline src/llama_stack_client/lib/cli/datasets/register.py:51]: kill these filler comments. I know where they come from :)

File: src/llama_stack_client/lib/cli/eval/run_scoring.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    if dataset_id is not None:
+        dataset = client.datasets.retrieve(dataset_id=dataset_id)
+        if not dataset:
+            click.BadParameter(
+                f"Dataset {dataset_id} not found. Please register using llama-stack-client datasets register"
+            )
+
+        # TODO: this will eventually be replaced with jobs polling from server vis score_bath
+        # For now, get all datasets rows via datasetio API
+        rows = client.datasetio.get_rows_paginated(
```
Comment (ashwinb) [inline src/llama_stack_client/lib/cli/eval/run_scoring.py:84]: nit: call it result?

**Review by ashwinb — APPROVED** [review]
lg!

File: src/llama_stack_client/lib/cli/eval/run_scoring.py (hunk 2 above)
Comment (yanxi0830) [reply src/llama_stack_client/lib/cli/eval/run_scoring.py]: done!

**Pull request merged**